### PR TITLE
New packages: google-crc32c and py-google-crc32c

### PR DESCRIPTION
**Pull request by marcusboden**
Created a package for google-crc32c and the python bindings for it.

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-google-crc32c/package.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+from spack import *
+
+
+class PyGoogleCrc32c(PythonPackage):
+    """This package wraps the google/crc32c hardware-based implementation
+    of the CRC32C hashing algorithm."""
+
+    homepage = "https://github.com/googleapis/python-crc32c"
+    git      = "https://github.com/googleapis/python-crc32c"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-google-crc32c/package.py]: It's better to download directly from PyPI so we can checksum:
```suggestion
   pypi = "google-crc32c/google-crc32c-1.3.0.tar.gz"
```
See https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html#pypi-vs-github

File: var/spack/repos/builtin/packages/google-crc32c/package.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class GoogleCrc32c(CMakePackage):
+    """CRC32C implementation with support for CPU-specific acceleration instructions."""
+
+    homepage = "https://github.com/google/crc32c"
+    git      = "https://github.com/google/crc32c"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/google-crc32c/package.py]: It's better to download via URL so we can checksum the tarball:
```suggestion
   url = "https://github.com/google/crc32c/archive/refs/tags/1.1.2.tar.gz"
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-google-crc32c/package.py (hunk 1 above)
Comment (marcusboden) [reply var/spack/repos/builtin/packages/py-google-crc32c/package.py]: Ah, I seem to have overlooked the tar.gz file on pypi. Fixed it now.

File: var/spack/repos/builtin/packages/google-crc32c/package.py (hunk 2 above)
Comment (marcusboden) [reply var/spack/repos/builtin/packages/google-crc32c/package.py]: Done.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**